### PR TITLE
Re-create the presync-only branch

### PR DESCRIPTION
**Pull request by stevenrbrandt**
The presync-only branch allows you to set presync_mode to "presync-only", and adds the logic for detecting and synchronizing before a function is called.

### CODE REVIEW

File: CarpetX/src/schedule.cxx [hunk 1]
```diff
@@ -1006,9 +1006,10 @@ int Initialise(tFleshConfig *config) {
   CCTKi_AddGH(config, 0, cctkGH);
 
   // Check presync mode
-  if (!CCTK_EQUALS(presync_mode, "mixed-error"))
-    CCTK_ERROR(
-        "CarpetX currently requires Cactus::presync_mode = \"mixed-error\"");
+  if (!CCTK_EQUALS(presync_mode, "mixed-error") &&
+      !CCTK_EQUALS(presync_mode, "presync-only"))
+    CCTK_ERROR("CarpetX currently requires Cactus::presync_mode = "
+               "\"mixed-error\" or \"presyc-only\"");
```
Comment (eschnett) [inline CarpetX/src/schedule.cxx:1012]: There's a typo "presyc"

File: CarpetX/src/schedule.cxx [hunk 2]
```diff
@@ -1716,6 +1717,27 @@ int CallFunction(void *function, cFunctionData *restrict attribute,
...
+        active_levels->loop([&](const auto &restrict leveldata) {
+          const auto &restrict groupdata = *leveldata.groupdata.at(rd.gi);
+          const valid_t &need = rd.valid;
+          valid_t have = groupdata.valid.at(rd.tl).at(rd.vi).get();
+          if (need.valid_ghosts && !have.valid_ghosts && have.valid_int)
+            sync_set.insert(rd.gi);
+        });
+      }
+    }
+    if (sync_set.size() > 0) {
```
Comment (eschnett) [inline CarpetX/src/schedule.cxx:1735]: Nitpicking: Better call `sync_set.empty()` instead of `size() > 0`.

**Review by eschnett — CHANGES_REQUESTED** [review]

File: CarpetX/src/schedule.cxx (hunk 1 above)
Comment (stevenrbrandt) [reply CarpetX/src/schedule.cxx]: Typo fixed.

File: CarpetX/src/schedule.cxx (hunk 2 above)
Comment (stevenrbrandt) [reply CarpetX/src/schedule.cxx]: Fixed.

File: CarpetX/src/schedule.cxx [hunk 3]
```diff
@@ -1006,9 +1006,10 @@ int Initialise(tFleshConfig *config) {
   CCTKi_AddGH(config, 0, cctkGH);
 
   // Check presync mode
-  if (!CCTK_EQUALS(presync_mode, "mixed-error"))
-    CCTK_ERROR(
-        "CarpetX currently requires Cactus::presync_mode = \"mixed-error\"");
+  if (!CCTK_EQUALS(presync_mode, "mixed-error") &&
+      !CCTK_EQUALS(presync_mode, "presync-only"))
+    CCTK_ERROR("CarpetX currently requires Cactus::presync_mode = "
+               "\"mixed-error\" or \"presnyc-only\"");
```
Comment (eschnett) [inline CarpetX/src/schedule.cxx:1012]: The typo is persnickety.

**Review by eschnett — APPROVED** [review]

**Pull request merged**